tools/reviewer/web/src/app/dashboard/reviews-panel/review/person-list/person-list.component.ts

```diff
@@ -1,97 +1,86 @@
+    this.title = this.isNeedAttention ? 'Reviewers' : 'CC';
+  }
 
   // To show editable fields
   showEditableProperty = false;
 
   // Following variable is used in editing the fields
-  propertyValue: string = '';
-  persons: string[] = [];
+  reviewerEmails: string = '';
+  reviewers: string[] = [];
 
   ngOnChanges() {
     this.getPropertyValue();
   }
 
   // Get property value from the Diff
   getPropertyValue(): void {
-    switch (this.property) {
-      case 'reviewers':
-        this.persons = this.diff.getReviewersList();
-        break;
-      case 'cc':
-        this.persons = this.diff.getCcList();
-        break;
-      default:
-        throw new Error('Unsupported property');
-    }
+    this.reviewers = this.diff.getReviewerList()
+      .filter(reviewer => {
+        return reviewer.getNeedsattention() === this.isNeedAttention;
+      })
+      .map(reviewer => reviewer.getEmail());
 
-    this.propertyValue = this.persons.join(', ');
+    this.reviewerEmails = this.reviewers.join(', ');
   }
 
   // Save the new value of property and update Diff
-  savePropertyValue(): void {
-    const persons = this.propertyValue.split(',');
-    // Trim white spaces
+  saveAttentionList(): void {
+    let reviewers: Reviewer[] = [];
 
-    // WARNING: Bad practice
-    // Do not use: object[propertyName]
-    // - It forces to use any.
-    // - It causes undetected errors.
-    // - It's difficult to maintain the code.
-    // use instead: object.field
-    // TODO: refactor this component ('switch' is not the best way too)
+    this.reviewerEmails
+      .split(',')
+      .map(reviewer => reviewer.trim())
+      .filter(reviewer => reviewer.length)
+      .forEach(email => {
+        const reviewer = new Reviewer();
+        reviewer.setEmail(email);
+        reviewer.setApproved(false);
+        reviewer.setNeedsattention(this.isNeedAttention);
+        reviewers.push(reviewer);
+      });
 
-    const newPersonsList = persons.map(v => v.trim()).filter(v => v.length);
-    let length;
+    const otherReviewers = this.diff.getReviewerList().filter(reviewer =>
+      reviewer.getNeedsattention() !== this.isNeedAttention
+    );
 
-    switch (this.property) {
-      case 'reviewers':
-        this.diff.setReviewersList(newPersonsList);
-        length = this.diff.getReviewersList().length;
-        break;
-      case 'cc':
-        this.diff.setReviewersList(newPersonsList);
-        length = this.diff.getReviewersList().length;
-        break;
-      default:
-        throw new Error('Unsupported property');
-    }
-
-    if (length === 0) {
-      this.onRemoveProperty.emit(this.property);
-    } else {
-      this.onUpdateDiff.emit(this.diff);
-    }
-  }
+    reviewers = reviewers.concat(otherReviewers);
+    this.diff.setReviewerList(reviewers);
 
-  // Save needAttentionOf list
-  saveAttentionList(name: string): void {
-    this.onAddToAttentionList.emit(name);
+    this.firebaseService.updateDiff(this.diff).subscribe(() => {
```

Comment:
We have the same code at other place as well.

Reply:
It's fine.
We have freedom to set success and error message in the future or add some pipe directly.
This is much more intuitive, then create additional method.

Reply:
Sorry, I mean that person-list shouldn't update the diff. Only review should update it and we can avoid calling this service method in a number of components.